### PR TITLE
test: add command line indicator option checking to fipsinstall

### PR DESCRIPTION
**Pull request by paulidale**
Validate that the relevant options are on when -pedantic is specified, off when it isn't and can be given to enable the setting.

- [ ] documentation is added or updated
- [x] tests are added or updated

### CODE REVIEW

File: test/recipes/03-test_fipsinstall.t
Comment (slontis) [inline test/recipes/03-test_fipsinstall.t:53]: padding will become pad due to another PR..

**Comment by paulidale** [comment]
Ping @openssl/otc @openssl/committers for reviews please.

**Review by nhorman — APPROVED** [review]
Can you link to the subsequent PR that changes padding to pad so we know to block it until this is merged please?

**Comment by paulidale** [comment]
It's #25162 which is still in draft.

**Review by slontis — APPROVED** [review]

**Comment by paulidale** [comment]
Kicking tests before merge.

**Pull request closed without merging**
